File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/lesson/LessonService.java
package kh.farrukh.sortmapperexamplejava.lesson;

import kh.farrukh.sortmapperexamplejava.lesson.model.LessonCreateRequestDTO;
import kh.farrukh.sortmapperexamplejava.lesson.model.LessonDetailsResponseDTO;
import kh.farrukh.sortmapperexamplejava.lesson.model.LessonResponseDTO;
import kh.farrukh.sortmapperexamplejava.lesson.model.LessonUpdateRequestDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface LessonService {

  void createLesson(LessonCreateRequestDTO requestDTO);

  LessonDetailsResponseDTO getLesson(Long id);

  Page<LessonResponseDTO> getLessons(Pageable pageable);

  void updateLesson(Long id, LessonUpdateRequestDTO requestDTO);

  void deleteLesson(Long id);

  LessonEntity findLesson(Long id);

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/lesson/LessonService.kt
package kh.farrukh.sortmapperexample.lesson

import kh.farrukh.sortmapperexample.lesson.model.LessonCreateRequestDTO
import kh.farrukh.sortmapperexample.lesson.model.LessonDetailsResponseDTO
import kh.farrukh.sortmapperexample.lesson.model.LessonResponseDTO
import kh.farrukh.sortmapperexample.lesson.model.LessonUpdateRequestDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface LessonService {

    fun createLesson(requestDTO: LessonCreateRequestDTO)

    fun getLessonById(id: Long): LessonDetailsResponseDTO

    fun getAllLessons(pageable: Pageable): Page<LessonResponseDTO>

    fun updateLesson(id: Long, requestDTO: LessonUpdateRequestDTO)

    fun deleteLesson(id: Long)

}